cmd/spiderpool-controller/cmd/daemon.go

```diff
@@ -445,6 +456,7 @@ func setupInformers() {
 		logger.Info("Begin to set up auto-created IPPool controller")
 		subnetAppController, err := applicationcontroller.NewSubnetAppController(
 			controllerContext.CRDManager.GetClient(),
+			controllerContext.CRDManager.GetAPIReader(),
 			controllerContext.SubnetManager,
 			applicationcontroller.SubnetAppControllerConfig{
 				EnableIPv4:                    controllerContext.Cfg.EnableIPv4,
```

Comment:
Overall, the code review indicates that there are several improvements and potential issues. Here are some observations:

- The addition of waiting for sync on manager cache is good, as controllers depend on having up-to-date information in the cache.
- The use of a goroutine for starting the GC Manager with error handling and context cancellation is a best practice for avoiding blocking main execution. However, it's important to note that the error channel should be buffered to avoid deadlock when Start() returns an error before reading from the channel.
- It seems that the `initSpiderControllerLeaderElect()` function was refactored to accept a context argument, which avoids sharing controllerContext.InnerCtx between different functions. This is good practice for managing the lifetime of context and avoiding bugs related to context cancellation.
- The changes made to the SubnetAppController constructor to include an APIReader dependency suggest that the code now also reads external resources besides the custom resources defined by the Controller Runtime. This is good if the app needs more information or to ensure consistency across their stateful objects.

Here are some suggestions for improvement:

- The use of Fatal() calls throughout the code prevents the application from continuing executing critical parts, even if other components might be functioning well. Since this is a daemon, it would be better to log errors and continue normal operation. Consider using Error() logging and returning err values to the caller.
- The configuration parameters seem to be globally shared among different modules, making it harder to change them later or introduce different environments, such as development or staging. Consider encapsulating the configuration options, either through structs or objects that can be initialized in different ways.
- There is no documentation for the code. Having comments explaining what each function, struct or variable does would be helpful for other developers who might work on this project later on.